src/libostree/ostree-kernel-args.c

```diff
@@ -804,3 +804,33 @@ ostree_kernel_args_get_last_value (OstreeKernelArgs *kargs, const char *key)
   const OstreeKernelArgsEntry *e = entries->pdata[entries->len-1];
   return _ostree_kernel_args_entry_get_value (e);
 }
+
+/**
+ * ostree_kernel_args_append_if_missing:
+ * @kargs: a OstreeKernelArgs instance
+ * @arg: key or key/value pair to be added
+ *
+ * Appends @arg which is in the form of key=value pair to the hash table kargs->table
+ * (appends to the value list if key is not in the hash table)
+ * and appends key to kargs->order if it is not in the hash table.
+ *
+ * Since: 2022.5
+ **/
+void
+ostree_kernel_args_append_if_missing (OstreeKernelArgs  *kargs,
+                                      const char *arg)
+{
+  g_autofree char *duped = g_strdup (arg);
+  const char *val = split_keyeq (duped);
+  if (val)
+    {
+      g_autofree char *key = g_strndup (duped, val - duped - 1);
+      duped = g_strdup (key);
```

Comment:
This looks like it's trying to parse out the key in key=val, but `split_keyeq()` already mutates the input by changing the `=` into a `\0`.